query/src/main/java/org/infinispan/query/impl/LifecycleManager.java

```diff
@@ -475,4 +485,31 @@ public void cacheManagerStarting(GlobalComponentRegistry gcr, GlobalConfiguratio
    public void cacheManagerStopped(GlobalComponentRegistry gcr) {
       mbeanServer = null;
    }
+
+   /**
+    * Sets {@link BooleanQuery#setMaxClauseCount} according to the value of {@link #MAX_BOOLEAN_CLAUSES_SYS_PROP} system
+    * property. This is executed only once, when first indexed cache is started.
+    */
+   private void setBooleanQueryMaxClauseCount() {
+      if (!maxBooleanClausesWasSet) {
+         maxBooleanClausesWasSet = true;
+         String maxClauseCountProp = SecurityActions.getSystemProperty(MAX_BOOLEAN_CLAUSES_SYS_PROP);
+         if (maxClauseCountProp != null) {
+            int maxClauseCount;
+            try {
+               maxClauseCount = Integer.parseInt(maxClauseCountProp);
+            } catch (NumberFormatException e) {
+               log.failedToParseSystemProperty(MAX_BOOLEAN_CLAUSES_SYS_PROP, e);
+               throw e;
```

Comment:
Could be prefer to launch an exception containing a configuration exception with the property name on it? Instead a numberformat.

Reply:
This is not a cache manager / cache configuration related error, it's a command line system property, so throwing a CacheConfigurationException would not be appropriate here. I would like to also avoid inventing a new exception class. Or do we already have one for such situations?